cclib/parser/molcasparser.py

```diff
@@ -279,6 +279,68 @@ def extract(self, inputfile, line):
                     self.logger.warning('File terminated before end of last SCF!')
                     break
 
+        # Harmonic frequencies in cm-1
+        #
+        # IR Intensities in km/mol
+        #
+        #                        1         2         3         4         5         6
+        #
+        #     Frequency:       i60.14    i57.39    128.18    210.06    298.24    309.65
+        #
+        #     Intensity:    3.177E-03 2.129E-06 4.767E-01 2.056E-01 6.983E-07 1.753E-07
+        #     Red. mass:      2.42030   2.34024   2.68044   3.66414   2.61721   3.34904
+        #
+        #     C1         x   -0.00000   0.00000   0.00000  -0.05921   0.00000  -0.06807
+        #     C1         y    0.00001  -0.00001  -0.00001   0.00889   0.00001  -0.02479
+        #     C1         z   -0.03190   0.04096  -0.03872   0.00001  -0.12398  -0.00002
+        #     C2         x   -0.00000   0.00001   0.00000  -0.06504   0.00000  -0.03487
+        #     C2         y    0.00000  -0.00000  -0.00000   0.01045   0.00001  -0.05659
+        #     C2         z   -0.03703  -0.03449  -0.07269   0.00000  -0.07416  -0.00001
+        #     C3         x   -0.00000   0.00001   0.00000  -0.06409  -0.00001   0.05110
+        #     C3         y   -0.00000   0.00001   0.00000   0.00152   0.00000  -0.03263
+        #     C3         z   -0.03808  -0.08037  -0.07267  -0.00001   0.07305   0.00000
+        if line[1:29] == 'Harmonic frequencies in cm-1':
+
+            self.skip_line(inputfile,'blank')
+            line = next(inputfile)
+
+            if 'IR' in line:
+                freqmode = 'IR'
+
+            while 'Thermochemistry' not in line:
+
+                if 'Frequency:' in line:
+                        if not hasattr(self, 'vibfreqs'):
```

Comment:
In `LogFile` there is a method called `set_attribute`, which is extremely useful but doesn't work for extending attributes like here. I was wondering whether you'd be up for udpating that method to work for extending things or adding another method for extending, whichever you think is more appropriate.

Reply:
Do we really need to update `set_attribute`. What are the problems associated with using built-in functions?
And if we really want to change the function, what exactly do we need it to do?

Reply:
Well, we've been very inconsistent with using `set_attribute` versus `hasattr`; we don't have good standards for it. The advantage to using `set_attribute` (when not `extend`ing a quantity) is that the logger will print a warning when overwriting an already set attribute.

Also, any code written using `extend` can be rewritten to use `append` if you add an intermediate temporary list.

Reply:
So maybe I can use `set_attribute` for assigning a value, but standard built-in functions for extending?

Reply:
Well, you don't have to do this. This is fine. I'm just suggesting this would be a nice thing to have. Consider it thinking out loud :)

I would say the main advantage of having our own attribute setters is that we can change them in the future, so it is an opportunity gain in my mind.

Now that I think about it, perhaps better and more Pythonic would be to just override __setattr__ for the object we set attributes on and do all of this transparently.